test/e2e/storage/testsuites/provisioning.go

```diff
@@ -340,15 +381,125 @@ func TestDynamicProvisioning(t StorageClassTest, client clientset.Interface, cla
 	return pv
 }
 
-func TestBindingWaitForFirstConsumer(t StorageClassTest, client clientset.Interface, claim *v1.PersistentVolumeClaim, class *storage.StorageClass) (*v1.PersistentVolume, *v1.Node) {
-	pvs, node := TestBindingWaitForFirstConsumerMultiPVC(t, client, []*v1.PersistentVolumeClaim{claim}, class)
+// PVWriteReadCheck checks that a PV retains data on a single node.
+//
+// It starts two pods:
+// - The first pod writes 'hello word' to the /mnt/test (= the volume) on one node.
+// - The second pod runs grep 'hello world' on /mnt/test on the same node.
+//
+// The node is selected by Kubernetes when scheduling the first
+// pod. It's then selected via its name for the second pod.
+//
+// If both succeed, Kubernetes actually allocated something that is
+// persistent across pods.
+//
+// This is a common test that can be called from a StorageClassTest.PvCheck.
+func PVWriteReadCheck(client clientset.Interface, claim *v1.PersistentVolumeClaim, volume *v1.PersistentVolume, node NodeSelection) {
```

Comment:
Can you rename this to PVWriteReadSingleNodeCheck to be more obvious about what the function is doing?

Reply:
Done.